lib/invisible_captcha/controller_ext.rb

```diff
@@ -22,6 +31,14 @@ def on_spam_action(options = {})
       end
     end
 
+    def on_threshold_spam_action(options = {})
+      if action = options[:on_threshold_spam]
+        send(action)
+      else
+        flash[:error] = InvisibleCaptcha.threshold_error_message
```

Comment:
If you redirected back would you not lose the values in the fields that you had filled in (if you were a user that just submitted the form too quickly)?